mautic/mautic
# Add "Send to DNC" permission for Trans Emails and option to send it in Campaign Action

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ❌ 
| New feature/enhancement? (use the a.x branch)      | ✅ 
| Deprecations?                          | ❌ 
| BC breaks? (use the c.x branch)        | ❌ 
| Automated tests included?              | ✅  
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



As a Mautic user, I want to see the "Send to DNC" option in Campaign's Transactional emails. When checked, the campaign email send should respect this value.

#### Steps to test this PR:
Check for:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Added “Send to DNC” checkbox to permissions to allow super users to assign who can mark emails as transactional/operational.
3. Insert a toggle button under email type in Campaigns that allows customers to send an email to contacts marked as DNC. Checkbox should say “Send email to those marked as Do Not Contact”. Grey out the checkbox if the user does not have the right permission.
4. Add a confirmation modal for customers when toggling “Warning - checking this box means that emails will be sent to any customer who meets the stated criteria, including those who unsubscribed.”.
5. Add a tooltip that states “Only users with the correct permission can mark emails as “Send to contacts marked as “Do Not Contact.” Tooltip should show regardless of permission status.
6. We will be sending emails to DNC contacts on the basis of new toggle button value.
